tooltester/src/main/java/io/dockstore/tooltester/client/cli/ReportCommand.java

```diff
@@ -0,0 +1,169 @@
+     */
+    private void getToolVersionTestResults(ToolVersion toolVersion, String runner, StatusReport report,
+            String toolId) {
+        String toolVersionId = toolVersion.getId();
+        String tag = toolVersion.getName();
+        String jenkinsJobName = buildName(runner, toolVersionId);
+        if (pipelineTester.getJenkinsJob(jenkinsJobName) == null) {
+            LOG.info("Could not get job: " + jenkinsJobName);
+            return;
+        }
+        int buildId = pipelineTester.getLastBuildId(jenkinsJobName);
+        if (buildId == 0 || buildId == -1) {
+            LOG.info("No build was ran for " + jenkinsJobName);
+            return;
+        }
+        PipelineNodeImpl[] pipelineNodes = pipelineTester.getBlueOceanJenkinsPipeline(jenkinsJobName);
+        // There's pretty much always going to be a "Parallel" node that does not matter
+        List<PipelineNodeImpl> filteredNodes = Arrays.stream(pipelineNodes)
+                .filter(pipelineNode -> !pipelineNode.getDisplayName().equals("Parallel") && pipelineNode.getDurationInMillis() > 0L)
+                .collect(Collectors.toList());
+
+        filteredNodes.forEach(pipelineNode -> {
+            try {
+                String state = pipelineNode.getState();
+                String result = pipelineNode.getResult();
+                if (state.equals(StateEnum.RUNNING.name())) {
+                    result = state;
+                }
+                String entity = pipelineTester.getEntity(pipelineNode.getLinks().getSteps().getHref());
+                String nodeLogURI = pipelineNode.getLinks().getSelf().getHref() + "log";
+                String longURL = jenkinsServerUrl + nodeLogURI;
+                String logURL = TinyUrl.getTinyUrl(longURL);
+                String logContent = pipelineTester.getEntity(nodeLogURI);
+                Gson gson = new Gson();
+                PipelineStepImpl[] pipelineSteps = gson.fromJson(entity, PipelineStepImpl[].class);
+                Long runtime = Arrays.stream(pipelineSteps).map(PipelineStepImpl::getDurationInMillis).mapToLong(Long::longValue).sum();
+                String date = pipelineNode.getStartTime();
+                String epochStartTime = TimeHelper.timeFormatToEpoch(date);
+                String duration = getDuration(state, date, runtime);
+                try {
+                    date = TimeHelper.timeFormatConvert(date);
+                } catch (ParseException e) {
+                    errorMessage("Could not parse start time " + date, CLIENT_ERROR);
+                }
+                List<String> record = Arrays
+                        .asList(date, toolVersion.getId(), tag, runner, pipelineNode.getDisplayName(), result, duration, logURL);
+                report.printAndWriteLine(record);
+                if (SEND_LOGS) {
+                    if (result.equals(StateEnum.SUCCESS.name())) {
+                        S3Client s3Client = new S3Client();
```

Comment:
Is it expensive to create an S3Client? If yes, would it make sense to just create this once outside the loop? If no, then ignore this comment.